src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -2,10 +2,10 @@
 
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<MachineT> {
```

Comment:
I see you used `MachineT` in a lot of places, what is it? We have `Machine` abstract class that we should use. Also it shouldn't be just `Machine` - you have to somehow use it with `extends` or `super` depending on the place you use it.